internal/pkg/cli/pipeline_deploy.go

```diff
@@ -143,24 +144,44 @@ func newDeployPipelineOpts(vars deployPipelineVars) (*deployPipelineOpts, error)
 				},
 			}
 		},
+		wsAppName: tryReadingAppName(),
 		svcBuffer: &bytes.Buffer{},
 		jobBuffer: &bytes.Buffer{},
 	}, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *deployPipelineOpts) Validate() error {
-	if o.name != "" {
-		if err := o.validatePipelineName(); err != nil {
-			return err
+	if o.wsAppName == "" {
+		return errNoAppInWorkspace
+	}
+	// The passed-in app name value must be the same as the workspace app name, as we need to be in the correct workspace to read the pipeline manifest.
+	if o.appName != "" {
+		if o.appName != o.wsAppName {
+			return fmt.Errorf("cannot specify app %s because the workspace is already registered with app %s", o.appName, o.wsAppName)
+		}
+		// Validate the app name.
+		if _, err := o.store.GetApplication(o.appName); err != nil {
+			return fmt.Errorf("get application %s configuration: %w", o.appName, err)
 		}
```

Comment:
Maybe validate the app probably when the flag is not set as well? In case the app specified within ws file doesn't have a SSM record

Reply:
Yeah, I thought about that. Then I considered that we haven't validated the result of `TryReadingAppName()` anywhere previously. But yes, in the spirit of improving erroring, I'll change it here.